astropy/io/fits/util.py

```diff
@@ -186,16 +184,13 @@ def itersubclasses(cls, _seen=None):
     D
     E
     C
-    >>> # get ALL (new-style) classes currently defined
+    >>> # get ALL classes currently defined
     >>> [cls.__name__ for cls in itersubclasses(object)]
     [...'tuple', ...'type', ...]
 
     From http://code.activestate.com/recipes/576949/
     """
 
-    if not isinstance(cls, type):
```

Comment:
This would also act if one passed in, e.g., an instance instead of a class... But since this is a private routine, I think it is OK to just remove it (as is, you get the error only when you really iterate, not when you call the function).

Reply:
Ah, I haven't thought about that. But the error message would have been misleading anyway. But a `TypeError` would probably be more appropriate than an `AttributeError`. 

Would you say I should include an `except AttributeError` clause?

Reply:
No need to worry too much given that it's only used internally in `io.fits`. Actually I wonder if this function is really needed, it's only used once, so it could be replaced by a direct iteration on `__subclasses__()` (and the helper function creates a cache that is not used after).